pkg/profiler/cpu/maps.go

```diff
@@ -185,15 +193,21 @@ func clearBpfMap(bpfMap *bpf.BPFMap) error {
 type bpfMaps struct {
 	logger log.Logger
 
-	module    *bpf.Module
-	byteOrder binary.ByteOrder
+	nativeModule *bpf.Module
+	rbperfModule *bpf.Module
+	byteOrder    binary.ByteOrder
 
 	debugPIDs *bpf.BPFMap
 
-	stackCounts      *bpf.BPFMap
-	stackTraces      *bpf.BPFMap
-	dwarfStackTraces *bpf.BPFMap
-	processInfo      *bpf.BPFMap
+	stackCounts                *bpf.BPFMap
+	stackTraces                *bpf.BPFMap
+	dwarfStackTraces           *bpf.BPFMap
+	interpreterStackTraces     *bpf.BPFMap
+	pidToRubyThread            *bpf.BPFMap
+	rubyVersionSpecificOffsets *bpf.BPFMap
+	frameTable                 *bpf.BPFMap
+	processInfo                *bpf.BPFMap
+	InterpreterFrames          map[uint32]string
```

Comment:
I think this shouldn't be a field. It should be instead returned from the method (please check my other comment). What do you think?

Reply:
I did this thinking on some use cases but completely forgot what it is, so maybe it was something silly and unfounded? 😄. Will change to what you suggested